Product/Production/Services/DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/docsubmission/audit/transform/DSDeferredResponseAuditTransforms.java

```diff
@@ -0,0 +1,242 @@
+        String serviceName, boolean isRequesting, Properties webContextProperties) {
+
+        String ipOrHost = isRequesting ? getDSDeferredRequestInitiatorAddress(target) : getLocalHostAddress();
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+        participant.setUserID(isRequesting ? NhincConstants.WSA_REPLY_TO
+            : getInboundReplyToFromHeader(webContextProperties));
+        if (isRequesting) {
+            participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
+        }
+        participant.setNetworkAccessPointID(ipOrHost);
+        participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(ipOrHost));
+        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE, null,
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME,
+            AuditTransformsConstants.ACTIVE_PARTICIPANT_ROLE_CODE_SOURCE_DISPLAY_NAME));
+        participant.setUserIsRequestor(Boolean.TRUE);
+        return participant;
+
+    }
+
+    @Override
+    protected AuditMessageType.ActiveParticipant getActiveParticipantDestination(NhinTargetSystemType target,
+        boolean isRequesting, Properties webContextProperties, String serviceName) {
+
+        String url;
+        String ipOrHost;
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+        url = isRequesting ? getWebServiceUrlFromRemoteObject(getLocalHCIDNhinTarget(),
+            NhincConstants.NHINC_XDR_RESPONSE_SERVICE_NAME) : getWebServiceUrlFromRemoteObject(
+                convertToNhinTarget(getMessageCommunityId(getAssertion(), target, isRequesting)),
+                NhincConstants.NHINC_XDR_RESPONSE_SERVICE_NAME);
+        if (url != null) {
+            try {
+                participant.setUserID(url);
+                ipOrHost = new URL(url).getHost();
+                participant.setNetworkAccessPointID(ipOrHost);
+                participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(ipOrHost));
+            } catch (MalformedURLException ex) {
+                LOG.error("Couldn't parse the given NetworkAccessPointID as a URL: " + ex.getLocalizedMessage(), ex);
+                // The url is null or not a valid url; for now, set the user id to anonymous
+                participant.setUserID(AuditTransformsConstants.ACTIVE_PARTICIPANT_USER_ID_SOURCE);
+                // TODO: For now, hardcode the value to localhost; need to find out if this needs to be set
+                participant.setNetworkAccessPointTypeCode(AuditTransformsConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_NAME);
+                participant.setNetworkAccessPointID(AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS);
+            }
+        }
```

Comment:
So you have default settingsfor userId and NetworkAccessPointId and NetworkAccessPointTypeCode, if an exception is thrown but not if the url is null.  why is that?